File: 02linear_list/01array/211_Remove_Duplicates_from_Sorted_Array/main.py
#!/usr/bin/env python
# coding=utf-8

#注意 python中没有++操作符 , a+=1 也不表示一个数

class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        n = len(A)
        if 0 == n:
            return 0;
        index = 0
        i = 1
        while i < n:
            if A[index] != A[i]:
                index += 1
                A[index] = A[i]
            i += 1

        return index+1

if __name__ == "__main__":

    s = Solution()
    A = [1,2,2,3,4,5,5,7]
    len = s.removeDuplicates(A)
    print len
    print A[:len]

File: 02linear_list/01array/211_Remove_Duplicates_from_Sorted_Array/main.cpp
/*************************************************************************
	> File Name: 211.cpp
	> Author:zhaoyihong 
	> Mail:zhaoyihong_at_126_dot_com 
    > Created Time: 2015年03月16日 星期一 13时36分05秒
 ************************************************************************/

#include  <iostream>
using namespace std;


class Solution {
    public:
        static int removeDuplicates(int A[], int n) 
        {
            if(n == 0 || A==nullptr) 
            {
                return 0;
            }
            
            int index = 0;
            for(int i=1;i<n;++i)
            {
                if(A[index] != A[i])
                {
                    A[++index] = A[i];
                }
            }
            return index+1;
        }                    
};

int main(void)
{
    int a[] = {1,2,2,3,4,5,5,5,7};
    int len = Solution::removeDuplicates(a,sizeof(a)/sizeof(*a));
    for(int i=0;i<len;++i)
    {
        cout << a[i] << " ";
    }
    cout << endl;
}
